src/main/blackbox/blackbox_io.c

```diff
@@ -470,7 +470,7 @@ bool blackboxDeviceOpen(void)
      *
      * 9 / 1250 = 7200 / 1000000
      */
-    blackboxWriteChunkSize = MAX((masterConfig.looptime * 9) / 1250, 4);
+    blackboxWriteChunkSize = MAX((GUARD_LOOPTIME * 9) / 1250, 4);
```

Comment:
Guys what about changing the looptime number into hz?

set looptime = 666 would mean 666hz

we can also change the name in set refreshrate or something else?

It would always be rounded to the closest possible value
Op 5 aug. 2015 11:55 schreef "Petr Ledvina" <notifications@github.com>:

> In src/main/blackbox/blackbox_io.c
> <https://github.com/cleanflight/cleanflight/pull/1186#discussion_r36285808>
> :
>
> > @@ -470,7 +470,7 @@ bool blackboxDeviceOpen(void)
> >       *
> >       * 9 / 1250 = 7200 / 1000000
> >       */
> > -    blackboxWriteChunkSize = MAX((masterConfig.looptime * 9) / 1250, 4);
> > +    blackboxWriteChunkSize = MAX((GUARD_LOOPTIME * 9) / 1250, 4);
>
> I'll make the heuristic overridable at least for same time, so that new
> users start with heuristic, but version upgrade will be closer to old
> behavior (with looptime rounded up to next available gyro sampling rate).
>
> Rule of thumb is that sampling time should ve 10x-100x system time
> constant, so there is no gain going over 1kHz looptime (but I may be wrong).
> Personally I place looptime into 250Hz range now, but it may be better to
> go a bit higher..
> I'am more interested in 'advanced' algorithms that will allow higher
> looptime then brute-forcing it ..
>
> BTW: attitude is now estimated at loop frequency, but ACC correction is
> filtered at 1/600.
> This is probably overkill, ACC correction may be done at much lower
> frequency (maybe using filtered ACC data) and just integrate gyro reading
> on each loop (and integration may be even improved if low-noise gyro rate
> derivative derivative is available). This will greatly reduce ACC penalty
> ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cleanflight/cleanflight/pull/1186/files#r36285808>.
>